store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9557,7 +9574,29 @@ public Document addDocumentRevision(OperationContext octxt, int docId, ParsedDoc
         } finally {
             endTransaction(success);
             sm.quietDelete(staged);
+            // when file revision is done check for acl object and update file size time and
+            // details
+            updateFileSharedWithMe(pd, doc);
+        }
+    }
+
+    public void updateFileSharedWithMe(ParsedDocument pd, Document doc) throws ServiceException {
+        if (doc instanceof Document && doc.getACL() != null) {
```

Comment:
add null check for pd & doc objects before use to avoid NPE

Reply:
Added the check